src/Neo/SmartContract/Native/PolicyContract.cs

```diff
@@ -132,40 +132,118 @@ internal override ContractTask InitializeAsync(ApplicationEngine engine, Hardfor
                 engine.SnapshotCache.Add(_maxValidUntilBlockIncrement, new StorageItem(engine.ProtocolSettings.MaxValidUntilBlockIncrement));
                 engine.SnapshotCache.Add(_maxTraceableBlocks, new StorageItem(engine.ProtocolSettings.MaxTraceableBlocks));
             }
+
+            // After Faun Hardfork the unit it's pico-gas, before it was datoshi
+
+            if (hardfork == Hardfork.HF_Faun)
+            {
+                // Add decimals to exec fee factor
+                var item = engine.SnapshotCache.TryGet(_execFeeFactor) ??
+                    throw new InvalidOperationException("Policy was not initialized");
+                item.Set((uint)(BigInteger)item * ApplicationEngine.FeeFactor);
+
+                // Add decimals to storage price
+                item = engine.SnapshotCache.TryGet(_storagePrice) ??
+                    throw new InvalidOperationException("Policy was not initialized");
+                item.Set((uint)(BigInteger)item * ApplicationEngine.FeeFactor);
+
+                // Add decimals to fee per Byte
+                item = engine.SnapshotCache.TryGet(_feePerByte) ??
```

Comment:
These 3 sets could be a function, there are too similar.

Reply:
I k now, it can be reduced if the argument it's `bool faunEnabled`, otherwise, it's different an ApplicationEngine than the ProtocolSettings